doc/nrf/ug_bootloader_adding.rst

```diff
@@ -0,0 +1,436 @@
+
+   Since each bootloader is built with its own signature key, using a different private key with an upgradable bootloader will not cause problems with the secure boot chain.
+   You can also use the same private key for both the immutable and upgradable bootloaders, as long as the key type is supported by both of them.
+
+.. _ug_bootloader_adding_presigned_variants:
+
+Pre-signed variants
+===================
+
+Enable the :option:`CONFIG_BUILD_S1_VARIANT` option when building the upgradable bootloader to automatically generate :ref:`pre-signed variants <upgradable_bootloader_presigned_variants>` of the image for both slots:
+
+.. code-block::
+
+   west build -b nrf52840dk_nrf52840 zephyr/samples/hello_world -- \
+   -DCONFIG_SECURE_BOOT=y \
+   -DCONFIG_BOOTLOADER_MCUBOOT=y \
+   -DCONFIG_BUILD_S1_VARIANT=y
+
+This is a necessary step for creating application update images for use with :ref:`ug_fw_update`.
+
+.. _ug_bootloader_adding_testing:
+
+Testing the boot chain
+**********************
+
+To test the bootloader(s) and the secure boot chain, build and program them with any sample as described above.
+
+By default, both |NSIB| and MCUboot print information to their serial output on boot.
+This output will include information about the validation of images in its slots, as well as firmware-specific information if using :option:`CONFIG_FW_INFO` with |NSIB|.
+To see this output:
+
+1. |connect_terminal|
+#. Reset the development kit.
+#. Observe that each bootloader in the chain prints its information upon boot (some values may vary by build):
+
+.. tabs::
+
+   .. tab:: |NSIB|
+
+      .. code-block::
+
+         Attempting to boot slot 0.
+         Attempting to boot from address 0x9000.
+         Verifying signature against key 0.
+         Hash: 0xc0...71
+         Firmware signature verified.
+         Firmware version 1
+         Setting monotonic counter (version: 1, slot: 0)
+
+   .. tab:: MCUboot
+
+      .. code-block::
+
+         [00:00:00.359,039] <inf> mcuboot: Starting bootloader
+         [00:00:00.365,295] <inf> mcuboot: Primary image: magic=unset, swap_type=0x1, copy_done=0x3, image_ok=0x3
+         [00:00:00.375,671] <inf> mcuboot: Secondary image: magic=unset, swap_type=0x1, copy_done=0x3, image_ok=0x3
+         [00:00:00.386,169] <inf> mcuboot: Boot source: none
+         [00:00:00.391,815] <inf> mcuboot: Swap type: none
+         [00:00:00.420,166] <inf> mcuboot: Bootloader chainload address offset: 0xc000
+         [00:00:00.428,039] <inf> mcuboot: Jumping to the first image slot
+
+When compiled with minimal configurations that disable logging output, such as ``prj_minimal.conf``, the bootloader information output can be disabled altogether or per-bootloader.
+Refer to the source code directories of each bootloader to see what minimal configuration options are already available, or add them through a custom Kconfig fragment if desired.
+
+.. _ug_bootloader_adding_fragments:
+
+Bootloader configurations and fragments
```

Comment:
The title of this section suggests a reference concept section, while in fact it is a task: Assigning overlays with project configurations. I strongly suggest rephrasing the title and rewriting the contents to a task-based format.

Reply:
fixed